.github/container/Dockerfile.triton

```diff
@@ -17,48 +18,62 @@ ENV TRITON_NVDISASM_PATH=/usr/local/cuda/bin/nvdisasm
 
 ###############################################################################
-## Check out Triton source and build a wheel
+## Check out Triton source
 ###############################################################################
-FROM base as builder
-ARG URLREF_OXLA_TRITON
-ARG SRC_PATH_OXLA_TRITON
-
-# bump-openxla-triton.sh ensures that the commit of openxla-triton referenced
-# in the manifest file is consistent with the commit of xla, and that any extra
-# patches are available under patches/openxla-triton
-RUN git-clone.sh ${URLREF_OXLA_TRITON} ${SRC_PATH_OXLA_TRITON}
+# Use XLA's Bazel configuration to get the relevant tag from the openxla/triton
+# fork's llvm-head branch and apply XLA's extra patches to it
+FROM ${BASE_IMAGE} as checkout
+ARG SRC_PATH_JAX
+ARG SRC_PATH_XLA
 RUN <<"EOF" bash -ex
-shopt -s nullglob
-cd /opt/openxla-triton
-for patch in /opt/manifest.d/patches/openxla-triton/*.patch; do
-  patch -p1 < "${patch}"
-done
-git diff
+pushd "${SRC_PATH_XLA}"
+BAZEL=$(find "${SRC_PATH_JAX}/build" -type f -executable -name 'bazel-*')
+"${BAZEL}" --output_base=/opt/triton-checkout fetch @triton//:BUILD
 EOF
 
+###############################################################################
+## Get (only) the Triton source from the checkout step
+###############################################################################
+FROM base as builder
+ARG SRC_PATH_TRITON
+ARG SRC_PATH_XLA
+COPY --from=checkout /opt/triton-checkout/external/triton "${SRC_PATH_TRITON}"
 RUN <<"EOF" bash -ex
-mkdir -p "${SRC_PATH_OXLA_TRITON}/dist"
-# This sidesteps the RTTI-related error in https://github.com/openai/triton/pull/3213 that otherwise breaks container
-# builds with openxla/triton tag cl608559313
-sed -i 's|^add_subdirectory(unittest)|# unit tests disabled|' "${SRC_PATH_OXLA_TRITON}/CMakeLists.txt"
-sed -i 's|backends = _copy_backends(\["nvidia", "amd"\])|backends = _copy_backends(["nvidia"])|g' "${SRC_PATH_OXLA_TRITON}/python/setup.py"
-sed -i '1s|^|include_directories(${CMAKE_SOURCE_DIR}/third_party/nvidia/backend/include)\n|' "${SRC_PATH_OXLA_TRITON}/lib/Conversion/TritonGPUToLLVM/CMakeLists.txt"
-pip wheel --wheel-dir="${SRC_PATH_OXLA_TRITON}/dist" "${SRC_PATH_OXLA_TRITON}/python"
+mkdir -p "${SRC_PATH_TRITON}/dist"
+sed -i -e 's|^add_subdirectory(unittest)|# unit tests disabled|' "${SRC_PATH_TRITON}/CMakeLists.txt"
+sed -i -e 's|BackendInstaller.copy(\["nvidia", "amd"\])|BackendInstaller.copy(["nvidia"])|g' "${SRC_PATH_TRITON}/python/setup.py"
+# Extra patches to Triton maintained in XLA. These are already applied in the checkout stage.
+XLA_TRITON_PATCHES="${SRC_PATH_XLA}/third_party/triton"
+# This patch only works with a newer LLVM than is pinned in the branch.
+PATCH_FILE="${XLA_TRITON_PATCHES}/llvm_integration/cl623185214.patch"
+if [[ $(cat "${SRC_PATH_TRITON}/cmake/llvm-hash.txt") == "6f44bb7717897191be25aa01161831c67cdf5b84" && -f "${PATCH_FILE}" ]]; then
```

Comment:
Can we store `6f44bb7717897191be25aa01161831c67cdf5b84` into a more declarative variable?